bindings/spt/solo5_spt.lds.in

```diff
@@ -12,7 +12,11 @@ SECTIONS {
         *(.text.*)
     }
 
+#if defined(__powerpc64__)
```

Comment:
Why is the different alignment (and all the associated rigmarole to preprocess the ldscript) required for PPC64?

Reply:
PPC64's mprotect protects memory in the steps of 0x10000 rather than 0x1000. We get memory access errors if we only write-protect for example 0x5000 bytes for some chunk and then immediately after that load the next chunk from the ELF. That's because it sets 0x10000 bytes to read-only rather than the requested 0x5000.

Reply:
Ah, so Linux/PPC64 uses a 64k page size by default, that is a crucial piece of information I was missing.

I don't have access to a PPC64 to test on (which would help), so can you try the following for me instead of your current ldscript changes? (i.e. using the original lds w/o preprocessing)

In the SPT ldscript, wherever it says

    . = ALIGN(0x1000);

change it to

    . = ALIGN(CONSTANT(MAXPAGESIZE));

Then, in `Makefile.common`, change `-z max-page-size=0x1000` to `-z max-page-size=0x10000`.

Ideally compare the output of `readelf -l test_hello.spt` before and after this change, they should be the same.

If this works, then I can use this approach in the ldscripts throughout (assuming it's portable to the LLVM linker) and gate it in a single place (configure.sh), passing the correct page size down to the Makefiles.

Reply:
Alignment is good after rebasing on your PR.